ambry-messageformat/src/test/java/com.github.ambry.messageformat/MessageFormatRecordTest.java

```diff
@@ -148,6 +130,77 @@ public void deserializeTest() {
   }
 
   @Test
+  public void testBlobPropertyV1() throws IOException, MessageFormatException {
+    // Test Blob property V1 Record
+    short[] versions = new short[]{BlobPropertiesSerDe.Version1, BlobPropertiesSerDe.Version2};
+    for (short version : versions) {
+      BlobProperties properties;
+      long blobSize = TestUtils.RANDOM.nextLong();
+      long ttl = TestUtils.RANDOM.nextInt();
+      if (version == BlobPropertiesSerDe.Version1) {
+        properties = new BlobProperties(blobSize, "id", "member", "test", true, ttl);
+      } else {
+        short accountId = Utils.getRandomShort(TestUtils.RANDOM);
+        short containerId = Utils.getRandomShort(TestUtils.RANDOM);
+        short issuerAccountId = Utils.getRandomShort(TestUtils.RANDOM);
+        properties =
+            new BlobProperties(blobSize, "id", "member", "test", true, ttl, accountId, containerId, issuerAccountId);
+      }
+      ByteBuffer stream;
+      if (version == BlobPropertiesSerDe.Version1) {
+        stream = ByteBuffer.allocate(getBlobPropertiesRecordSize(properties));
+        MessageFormatRecord.BlobProperties_Format_V1.serializeBlobPropertiesRecord(stream, properties);
+      } else {
+        stream = ByteBuffer.allocate(getBlobPropertiesV2RecordSize(properties));
+        serializeBlobPropertiesV2Record(stream, properties);
+      }
+      stream.flip();
+      BlobProperties result = MessageFormatRecord.deserializeBlobProperties(new ByteBufferInputStream(stream));
+      Assert.assertEquals(properties.getBlobSize(), result.getBlobSize());
+      Assert.assertEquals(properties.getContentType(), result.getContentType());
+      Assert.assertEquals(properties.getCreationTimeInMs(), result.getCreationTimeInMs());
+      Assert.assertEquals(properties.getOwnerId(), result.getOwnerId());
+      Assert.assertEquals(properties.getServiceId(), result.getServiceId());
+      // @TODO: fix this once BlobProperty V2 is enabled
+      if (version == BlobPropertiesSerDe.Version1) {
```

Comment:
yes, it works for now. But when we start writing in version2, no one can write in version1 anymore.